File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        vector<int> ans;
        set<int> s1,s2;
        for(int i:nums1)
            s1.insert(i);
        for(int i:nums2)
            s2.insert(i);
        for(auto i:s1)
        {
            if(s2.count(i))
                ans.push_back(i);
        }
        return ans;
    }
};

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    private int[] helper(Set<Integer> set1, Set<Integer> set2)
    {
        int[] res = new int[set1.size()];
        int k =0;
        for(Integer s: set1)
        {
            if(set2.contains(s))
               {
                   res[k++] = s;
               }}
               return Arrays.copyOf(res, k);
    }
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> s1 = new HashSet<>();
        for(Integer n : nums1)
            s1.add(n);
        Set<Integer> s2 = new HashSet<>();
        for(Integer n : nums2)
            s2.add(n);
        if(s1.size()<s2.size())
            return helper(s1,s2);
        else
            return helper(s2,s1);
    }
}